### ISSUE TITLE
[bionic, qtwayland5] the second time we send configure event to a wl_shell_surface hangs the client

### ISSUE THREAD

**Issue by AlanGriffiths** [MirServer/mir#197]
One way to reproduce:

$ miral-app

<Alt+F11> to maximise the window
check it is responsive
<Alt+F11> to unmaximise the window

expect: still responsive
actual: not responsive

Can also reproduce by drag-resize, but it's hard to see that it's the second configure that causes the hang.

**Comment by AlanGriffiths** [comment]
The first "configure":

```
[476694.126] wl_shell_surface@19.configure(10, 1280, 1024)
[2018-02-27 15:16:29.613628] miral::Window Management: advise_state_change window_info={name=alan@X1c3-octopull: ~/display_server/mir, type=freestyle, state=restored, size=(1280, 1024), children={}, preferred_orientation=0xf, confine_pointer=0, output_id=0}, state=maximized
[2018-02-27 15:16:29.614085] miral::Window Management: ====
[476698.256]  -> wl_shm_pool@27.destroy()
[476698.289]  -> wl_buffer@28.destroy()
[476698.466]  -> wl_shm@10.create_pool(new id wl_shm_pool@33, fd 16, 5242880)
[476698.513]  -> wl_shm_pool@33.create_buffer(new id wl_buffer@35, 0, 1280, 1024, 5120, 0)
[476713.964]  -> wl_surface@15.frame(new id wl_callback@29)
[476714.003]  -> wl_surface@15.attach(wl_buffer@35, 0, 0)
[476714.028]  -> wl_surface@15.damage(3, 30, 1274, 991)
[476714.061]  -> wl_surface@15.commit()
[476714.096]  -> wl_surface@15.frame(new id wl_callback@34)
[476714.118]  -> wl_surface@15.attach(wl_buffer@35, 0, 0)
[476714.139]  -> wl_surface@15.damage(3, 30, 1274, 991)
[476714.161]  -> wl_surface@15.commit()
[476714.173]  -> wl_surface@15.frame(new id wl_callback@32)
[476714.187]  -> wl_surface@15.attach(wl_buffer@35, 0, 0)
[476714.206]  -> wl_surface@15.damage(3, 30, 1274, 991)
[476714.230]  -> wl_surface@15.commit()
[476714.275] wl_shm_pool@27.destroy()
[476714.292]  -> wl_display@1.delete_id(27)
[476714.301] wl_buffer@28.destroy()
...
```

The second "configure":
```
[790752.883] wl_shell_surface@19.configure(10, 646, 513)
[2018-02-27 15:21:43.671458] miral::Window Management: advise_state_change window_info={name=alan@X1c3-octopull: ~/display_server/mir, type=freestyle, state=maximized, size=(646, 513), restore_rect=((317, 170), (646, 513)), children={}, preferred_orientation=0xf, confine_pointer=0, output_id=0}, state=restored
[2018-02-27 15:21:43.671655] miral::Window Management: ====
[790755.172]  -> wl_shm_pool@26.destroy()
[790755.294]  -> wl_buffer@23.destroy()
[790755.830]  -> wl_shm@10.create_pool(new id wl_shm_pool@25, fd 16, 1325592)
[790755.892]  -> wl_shm_pool@25.create_buffer(new id wl_buffer@24, 0, 646, 513, 2584, 0)
[790763.619]  -> wl_surface@15.frame(new id wl_callback@29)
[790763.638]  -> wl_surface@15.attach(wl_buffer@24, 0, 0)
[790763.656]  -> wl_surface@15.damage(3, 30, 640, 480)
[790763.679]  -> wl_surface@15.commit()
[790763.699]  -> wl_surface@15.frame(new id wl_callback@22)
[790763.716]  -> wl_surface@15.attach(wl_buffer@24, 0, 0)
[790763.736]  -> wl_surface@15.damage(3, 30, 640, 480)
[790763.762]  -> wl_surface@15.commit()
[790763.775]  -> wl_surface@15.frame(new id wl_callback@28)
[790763.791]  -> wl_surface@15.attach(wl_buffer@24, 0, 0)
[790763.812]  -> wl_surface@15.damage(3, 30, 640, 480)
[790763.840]  -> wl_surface@15.commit()
[790763.898] wl_shm_pool@26.destroy()
[790763.919]  -> wl_display@1.delete_id(26)
[790763.930] wl_buffer@23.destroy()
...
```

**Comment by AlanGriffiths** [comment]
This happens with both wl_shell and xdg-shell

**Comment by AlanGriffiths** [comment]
Interestingly on Fedora 27 this seems to be working just fine.

```
$ dnf info qt5-qtwayland
Version       : 5.9.4
```

On bionic we have:

```
apt show qtwayland5
Version: 5.9.3-0ubuntu1
```

That makes me suspect the problem is fixed in qtwayland 5.9.4.

**Comment by AlanGriffiths** [comment]
Yep. after 5.9.4 this stopped happening
